base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        store=True,
```

Comment:
Well my thought with the access rules is as follows:

* User A is assigned to Company B
* Product A is assigned to company_id A and company_ids A,B
* User A navigates to Product A
* Security issue when displaying company_id, which is included in the product view